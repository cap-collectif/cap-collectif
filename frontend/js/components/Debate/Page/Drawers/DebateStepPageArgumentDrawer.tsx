import * as React from 'react'
import { graphql, useFragment } from 'react-relay'
import { FormattedMessage, useIntl } from 'react-intl'
import { useDisclosure } from '@liinkiing/react-hooks'
import { Flex, Text, Button, Tag, Heading, Icon, CapUIIcon, CapUIFontSize } from '@cap-collectif/ui'
import DetailDrawer from '~ds/DetailDrawer/DetailDrawer'
import LoginOverlay from '~/components/Utils/LoginOverlay'
import { voteForArgument } from '~/components/Debate/ArgumentCard/ArgumentCard'
import type { DebateStepPageArgumentDrawer_argument$key } from '~relay/DebateStepPageArgumentDrawer_argument.graphql'
import type { DebateStepPageArgumentDrawer_viewer$key } from '~relay/DebateStepPageArgumentDrawer_viewer.graphql'
import ModalReportArgumentMobile from '~/components/Debate/Page/Arguments/ModalReportArgumentMobile'
import ModalArgumentAuthorMenu from '~/components/Debate/Page/Arguments/ModalArgumentAuthorMenu'
import ModalModerateArgumentMobile from '~/components/Debate/Page/Arguments/ModalModerateArgumentMobile'
import { useDebateStepPage } from '~/components/Debate/Page/DebateStepPage.context'
type Props = {
  readonly isOpen: boolean
  readonly onClose?: () => void
  readonly argument: DebateStepPageArgumentDrawer_argument$key
  readonly viewer: DebateStepPageArgumentDrawer_viewer$key | null | undefined
}
const ARGUMENT_FRAGMENT = graphql`
  fragment DebateStepPageArgumentDrawer_argument on DebateArgument
  @argumentDefinitions(isAuthenticated: { type: "Boolean!" }) {
    id
    author {
      username
    }
    type
    body
    viewerCanReport @include(if: $isAuthenticated)
    viewerDidAuthor @include(if: $isAuthenticated)
    viewerHasVote @include(if: $isAuthenticated)
    votes(first: 0) {
      totalCount
    }
    ...ModalModerateArgumentMobile_argument @include(if: $isAuthenticated)
    ...ModalArgumentAuthorMenu_argument @include(if: $isAuthenticated)
    ...ModalReportArgumentMobile_argument @include(if: $isAuthenticated)
  }
`
const VIEWER_FRAGMENT = graphql`
  fragment DebateStepPageArgumentDrawer_viewer on User {
    isAdmin
  }
`

const DebateStepPageArgumentDrawer = ({
  argument: argumentFragment,
  viewer: viewerFragment,
  ...drawerProps
}: Props): JSX.Element => {
  const argument = useFragment(ARGUMENT_FRAGMENT, argumentFragment)
  const viewer = useFragment(VIEWER_FRAGMENT, viewerFragment)
  const intl = useIntl()
  const { isOpen, onOpen, onClose } = useDisclosure(false)
  const isViewerAdmin = viewer && viewer.isAdmin
  const { widget, stepClosed } = useDebateStepPage()
  if (!argument) return null
  const isAuthor = argument.viewerDidAuthor
  return (
    <DetailDrawer {...drawerProps}>
      <DetailDrawer.Header textAlign="center" justifyContent="space-between">
        <Flex direction="column" spacing={1} flex={1}>
          <Text fontWeight="bold">
            {argument.author?.username ??
              intl.formatMessage({
                id: 'global.anonymous',
              })}
          </Text>

          <Tag variantColor={argument.type === 'FOR' ? 'green' : 'red'} interactive={false} alignSelf="center">
            <Heading as="h5" fontWeight="700" uppercase>
              <FormattedMessage id={argument.type === 'FOR' ? 'global.for' : 'global.against'} />
            </Heading>
          </Tag>
        </Flex>

        {argument.viewerCanReport && !stepClosed && (
          <Button
            rightIcon={CapUIIcon.More}
            aria-label={intl.formatMessage({
              id: 'global.menu',
            })}
            color="gray.500"
            onClick={onOpen}
          />
        )}

        {isViewerAdmin && !isAuthor && <ModalModerateArgumentMobile argument={argument} />}

        {isAuthor && !stepClosed && <ModalArgumentAuthorMenu argument={argument} />}
      </DetailDrawer.Header>

      <DetailDrawer.Body>
        <Text>{argument.body}</Text>
      </DetailDrawer.Body>

      <Flex
        direction="row"
        align="center"
        justify="center"
        position="fixed"
        bottom={0}
        bg="white"
        borderTopLeftRadius="16px"
        borderTopRightRadius="16px"
        width="100%"
        py={4}
      >
        <LoginOverlay enabled={!stepClosed} placement="bottom">
          <Button
            color="neutral-gray.500"
            leftIcon={<Icon name={argument.viewerHasVote ? 'CLAP' : 'CLAP_O'} size="lg" />}
            onClick={() =>
              voteForArgument(argument.id, argument.viewerHasVote, intl, argument.votes.totalCount, widget.location)
            }
            aria-label={intl.formatMessage({
              id: argument.viewerHasVote ? 'global.cancel' : 'vote.add',
            })}
            disabled={stepClosed}
            variant="link"
            variantColor="hierarchy"
          />
        </LoginOverlay>
        <Text ml={1} as="span" fontSize={CapUIFontSize.Headline} color="neutral-gray.900">
          {argument.votes.totalCount}
        </Text>
      </Flex>

      <ModalReportArgumentMobile show={isOpen} argument={argument} onClose={onClose} />
    </DetailDrawer>
  )
}

export default DebateStepPageArgumentDrawer
