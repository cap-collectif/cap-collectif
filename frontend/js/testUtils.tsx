import * as React from 'react'
import ReactDOM from 'react-dom'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { Provider as ReakitProvider } from 'reakit'
import { IntlProvider } from 'react-intl-redux'
import { RelayEnvironmentProvider } from 'react-relay'
import { CapUIProvider } from '@cap-collectif/ui'
import { BrowserRouter as Router } from 'react-router-dom'
import { theme } from '~/styles/theme'
import appStore from '~/stores/AppStore'
import type { FeatureFlagType } from '~relay/useFeatureFlagQuery.graphql'
import { FormProvider, useForm } from 'react-hook-form'

export const mockRandomValues = () => {
  global.Math.random = () => 0.5
}
export const enableFeatureFlags = (flags: Array<FeatureFlagType>) => {
  global.mockFeatureFlag.mockImplementation((flag: Array<FeatureFlagType>) => {
    if (flags.includes(flag)) {
      return true
    }

    return false
  })
}
export const disableFeatureFlags = () => {
  global.mockFeatureFlag.mockImplementation(() => false)
}
export const mockUrl = (url: string) => {
  // https://stackoverflow.com/questions/54021037/how-to-mock-window-location-href-with-jest-vuejs
  delete window.location
  // @ts-expect-error assign new URL(...) to window.location because property location is not writable
  window.location = new URL(url)
}
export const addsSupportForPortals = () => {
  // See: https://github.com/facebook/react/issues/11565
  // @ts-expect-error
  ReactDOM.createPortal = jest.fn(element => {
    return element
  })
}
export const clearSupportForPortals = () => {
  // @ts-expect-error
  ReactDOM.createPortal.mockClear()
}
type Props = {
  children: JSX.Element | JSX.Element[] | string
  store?: any
  useCapUIProvider?: boolean
}
export const MockProviders = ({ children, store = {}, useCapUIProvider = false }: Props) => {
  return (
    <Provider store={appStore(store)}>
      <Router>
        <IntlProvider>
          <ReakitProvider>
            {useCapUIProvider ? (
              <CapUIProvider>{children}</CapUIProvider>
            ) : (
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            )}
          </ReakitProvider>
        </IntlProvider>
      </Router>
    </Provider>
  )
}
type RelaySuspensFragmentTestProps = Props & {
  environment: any
}
export const RelaySuspensFragmentTest = ({
  children,
  environment,
  store = {},
  useCapUIProvider = false,
}: RelaySuspensFragmentTestProps) => {
  return (
    <MockProviders store={store} useCapUIProvider={useCapUIProvider}>
      <RelayEnvironmentProvider environment={environment}>
        <React.Suspense fallback="">{children}</React.Suspense>
      </RelayEnvironmentProvider>
    </MockProviders>
  )
}

export const FormWrapper = props => {
  const formMethods = useForm()

  return <FormProvider {...formMethods}>{props.children}</FormProvider>
}

export default MockProviders
