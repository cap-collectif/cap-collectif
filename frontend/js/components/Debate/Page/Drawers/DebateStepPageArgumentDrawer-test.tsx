/* eslint-env jest */
import * as React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { graphql, useLazyLoadQuery } from 'react-relay'
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils'
import { RelaySuspensFragmentTest, addsSupportForPortals, clearSupportForPortals, mockRandomValues } from '~/testUtils'
import type { DebateStepPageArgumentDrawerTestQuery } from '~relay/DebateStepPageArgumentDrawerTestQuery.graphql'
import DebateStepPageArgumentDrawer from './DebateStepPageArgumentDrawer'

describe('<DebateStepPageArgumentDrawer />', () => {
  let environment
  let testComponentTree
  let TestComponent
  let onClose
  const baseArgument = {
    id: 'debateArgument1',
    type: 'FOR',
    body: 'coucou',
    viewerCanReport: true,
    viewerDidAuthor: false,
    viewerHasVote: false,
    votes: {
      totalCount: 10,
    },
  }
  const defaultMockResolvers = {
    DebateArgument: () => baseArgument,
    User: () => ({
      isAdmin: false,
      username: 'Agui le penseur',
      media: {
        url: '<media-url>',
      },
      biography: 'Jsuis né dans les favela au brésil',
    }),
  }
  const query = graphql`
    query DebateStepPageArgumentDrawerTestQuery($id: ID = "<default>", $isAuthenticated: Boolean!)
    @relay_test_operation {
      argument: node(id: $id) {
        ...DebateStepPageArgumentDrawer_argument @arguments(isAuthenticated: $isAuthenticated)
      }
      viewer {
        ...DebateStepPageArgumentDrawer_viewer
      }
    }
  `
  afterEach(() => {
    clearSupportForPortals()
  })
  beforeEach(() => {
    // TODO: there is a random id but I can't find where
    mockRandomValues()
    addsSupportForPortals()
    environment = createMockEnvironment()
    onClose = jest.fn()

    const TestRenderer = props => {
      const data = useLazyLoadQuery<DebateStepPageArgumentDrawerTestQuery>(query, {
        isAuthenticated: true,
      })
      if (!data.argument) return null
      return <DebateStepPageArgumentDrawer argument={data.argument} viewer={data.viewer} {...props} />
    }

    TestComponent = props => (
      <RelaySuspensFragmentTest environment={environment}>
        <TestRenderer {...props} />
      </RelaySuspensFragmentTest>
    )

    environment.mock.queueOperationResolver(operation => MockPayloadGenerator.generate(operation, defaultMockResolvers))
  })
  it('should render when open', async () => {
    testComponentTree = ReactTestRenderer.create(<TestComponent isOpen onClose={onClose} />)
    expect(testComponentTree).toMatchSnapshot()
  })
  it('should render when open and viewer is the author', async () => {
    environment.mock.queueOperationResolver(operation =>
      MockPayloadGenerator.generate(operation, {
        ...defaultMockResolvers,
        DebateArgument: () => ({ ...baseArgument, viewerCanReport: false, viewerDidAuthor: true }),
      }),
    )
    testComponentTree = ReactTestRenderer.create(<TestComponent isOpen onClose={onClose} />)
    expect(testComponentTree).toMatchSnapshot()
  })
  it('should render when open and viewer has vote', async () => {
    environment.mock.queueOperationResolver(operation =>
      MockPayloadGenerator.generate(operation, {
        ...defaultMockResolvers,
        DebateArgument: () => ({ ...baseArgument, viewerHasVote: true }),
      }),
    )
    testComponentTree = ReactTestRenderer.create(<TestComponent isOpen onClose={onClose} />)
    expect(testComponentTree).toMatchSnapshot()
  })
  it('should render nothing when closed', async () => {
    testComponentTree = ReactTestRenderer.create(<TestComponent isOpen={false} onClose={onClose} />)

    expect(testComponentTree.toJSON().props.className.includes('toasts-container')).toBe(true)
    expect(testComponentTree.toJSON().children).toBeNull()
  })
})
