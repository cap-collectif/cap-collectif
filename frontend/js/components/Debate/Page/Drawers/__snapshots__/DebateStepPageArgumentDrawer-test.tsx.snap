// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebateStepPageArgumentDrawer /> should render when open 1`] = `
Array [
  <div
    className="sc-bcXGCL cgAMhb toasts-container toasts-container--top"
  />,
  <div
    data-testid="AnimatePresence"
  >
    <div
      className="sc-bcXGCL iLDPSx DetailDrawer__DetailDrawerInner-iBGoKl"
      style={
        Object {
          "opacity": 0,
          "transform": "translateX(100vw) translateZ(0)",
        }
      }
    >
      <div
        className="sc-bcXGCL ahQii"
      >
        <svg
          aria-hidden="true"
          className="sc-bcXGCL sc-eJDTud bPmEkv cpMilg cap-icon DetailDrawerHeader___StyledIcon-dNiWSR igkLcM detail__drawer--back-arow"
          fill="none"
          onClick={[MockFunction]}
          viewBox="0 0 12 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3H1M2.667 4.667L1 3l1.667-1.667"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1.5}
          />
        </svg>
        <div
          className="sc-bcXGCL gDgUrs"
        >
          <p
            className="sc-bcXGCL cUtTDY cap-text"
            style={
              Object {
                "marginLeft": 0,
                "marginTop": 0,
              }
            }
          >
            Agui le penseur
          </p>
          <div
            className="sc-bcXGCL kcdSuz sc-fXqqsB gKbWOe cap-tag"
            title=""
          >
            <h5
              className="sc-bcXGCL dUdqFq cap-heading"
            >
              global.for
            </h5>
          </div>
        </div>
      </div>
      <div
        className="sc-bcXGCL byWdSB"
        onClose={[MockFunction]}
      >
        <p
          className="sc-bcXGCL jMALnA cap-text"
        >
          coucou
        </p>
      </div>
      <div
        className="sc-bcXGCL cGdsLR"
        onClose={[MockFunction]}
      >
        <button
          aria-label="vote.add"
          className="sc-bcXGCL sc-laZSpB dDVxRe bGltjj cap-button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-eJDTud kAyhOl cpMilg cap-icon"
            fill="none"
            viewBox="0 0 10 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M.001 4.365c.055.941.05 1.722.046 2.41-.01 1.826-.027 3.016 1.242 4.3a3.107 3.107 0 004.425 0l3.441-3.481a1.063 1.063 0 00-.288-1.698l.57-.577a1.065 1.065 0 000-1.492 1.037 1.037 0 00-.323-.222 1.063 1.063 0 00-.473-1.483A1.063 1.063 0 008.5.806c-.34-.344-.857-.4-1.255-.17A1.036 1.036 0 005.55.308c-.02.021-1.84 1.854-3.004 3.03L2.368 2.06a1.048 1.048 0 00-1.042-1.028A1.048 1.048 0 00.283 2.09c.028.478.043.96.048 1.502a1.05 1.05 0 00-.33.773zm8.663 2.732l-3.442 3.48c-.949.96-2.493.96-3.441 0C.726 9.511.73 8.598.742 6.78c.005-.696.01-1.483-.046-2.44a.348.348 0 01.593-.227c.118.12.073.17.105.298v.002l.28 1.996c.04.286.387.404.59.2.6-.608-.063.062 3.46-3.5a.345.345 0 01.481.01.354.354 0 010 .497L4.73 5.107a.354.354 0 000 .498.345.345 0 00.492 0l1.967-1.99a.345.345 0 01.491 0 .354.354 0 010 .498L5.714 6.102a.354.354 0 000 .497.345.345 0 00.491 0L7.68 5.107a.345.345 0 01.492 0 .355.355 0 010 .498L6.697 7.097a.354.354 0 000 .497.345.345 0 00.492 0l.985-.997a.345.345 0 01.49.002.354.354 0 010 .498zm-.21-4.264a.354.354 0 010 .497l-.078.08a1.056 1.056 0 00-.492-.498l.079-.08a.345.345 0 01.491 0zm.492 1.989l-.079.08a1.05 1.05 0 00-.492-.498l.08-.08a.345.345 0 01.491 0 .355.355 0 010 .498zm-.937-3.519a.354.354 0 010 .496c-.043.045.257-.257-1.108 1.113a1.056 1.056 0 00-.49-.495l1.106-1.114a.345.345 0 01.492 0zm-5.463 3.03C3.584 3.283 6.018.831 6.042.806a.345.345 0 01.492 0c.13.131.135.337.02.474-3.25 3.282 1.53-1.553-4.29 4.333L2.09 4.365c.14.107.334.092.456-.032zM.98 2.066a.348.348 0 01.592-.228c.118.12.073.17.106.298l.228 1.63a1.037 1.037 0 00-.882-.458c-.007-.438-.021-.84-.044-1.242z"
              fill="currentColor"
            />
          </svg>
        </button>
        <span
          className="sc-bcXGCL hmaIsl cap-text"
        >
          10
        </span>
      </div>
      <div
        alwaysOpenInPortal={true}
        ariaLabel="global.report.submit"
        data-testid="Modal"
        hideCloseButton={true}
        onClose={[Function]}
        show={false}
        variantSize="xl"
      />
    </div>
  </div>,
]
`;

exports[`<DebateStepPageArgumentDrawer /> should render when open and viewer has vote 1`] = `
Array [
  <div
    className="sc-bcXGCL cgAMhb toasts-container toasts-container--top"
  />,
  <div
    data-testid="AnimatePresence"
  >
    <div
      className="sc-bcXGCL iLDPSx DetailDrawer__DetailDrawerInner-iBGoKl"
      style={
        Object {
          "opacity": 0,
          "transform": "translateX(100vw) translateZ(0)",
        }
      }
    >
      <div
        className="sc-bcXGCL ahQii"
      >
        <svg
          aria-hidden="true"
          className="sc-bcXGCL sc-eJDTud bPmEkv cpMilg cap-icon DetailDrawerHeader___StyledIcon-dNiWSR igkLcM detail__drawer--back-arow"
          fill="none"
          onClick={[MockFunction]}
          viewBox="0 0 12 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3H1M2.667 4.667L1 3l1.667-1.667"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1.5}
          />
        </svg>
        <div
          className="sc-bcXGCL gDgUrs"
        >
          <p
            className="sc-bcXGCL cUtTDY cap-text"
            style={
              Object {
                "marginLeft": 0,
                "marginTop": 0,
              }
            }
          >
            Agui le penseur
          </p>
          <div
            className="sc-bcXGCL kcdSuz sc-fXqqsB gKbWOe cap-tag"
            title=""
          >
            <h5
              className="sc-bcXGCL dUdqFq cap-heading"
            >
              global.for
            </h5>
          </div>
        </div>
      </div>
      <div
        className="sc-bcXGCL byWdSB"
        onClose={[MockFunction]}
      >
        <p
          className="sc-bcXGCL jMALnA cap-text"
        >
          coucou
        </p>
      </div>
      <div
        className="sc-bcXGCL cGdsLR"
        onClose={[MockFunction]}
      >
        <button
          aria-label="vote.add"
          className="sc-bcXGCL sc-laZSpB dDVxRe bGltjj cap-button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-eJDTud kAyhOl cpMilg cap-icon"
            fill="none"
            viewBox="0 0 10 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M.001 4.365c.055.941.05 1.722.046 2.41-.01 1.826-.027 3.016 1.242 4.3a3.107 3.107 0 004.425 0l3.441-3.481a1.063 1.063 0 00-.288-1.698l.57-.577a1.065 1.065 0 000-1.492 1.037 1.037 0 00-.323-.222 1.063 1.063 0 00-.473-1.483A1.063 1.063 0 008.5.806c-.34-.344-.857-.4-1.255-.17A1.036 1.036 0 005.55.308c-.02.021-1.84 1.854-3.004 3.03L2.368 2.06a1.048 1.048 0 00-1.042-1.028A1.048 1.048 0 00.283 2.09c.028.478.043.96.048 1.502a1.05 1.05 0 00-.33.773zm8.663 2.732l-3.442 3.48c-.949.96-2.493.96-3.441 0C.726 9.511.73 8.598.742 6.78c.005-.696.01-1.483-.046-2.44a.348.348 0 01.593-.227c.118.12.073.17.105.298v.002l.28 1.996c.04.286.387.404.59.2.6-.608-.063.062 3.46-3.5a.345.345 0 01.481.01.354.354 0 010 .497L4.73 5.107a.354.354 0 000 .498.345.345 0 00.492 0l1.967-1.99a.345.345 0 01.491 0 .354.354 0 010 .498L5.714 6.102a.354.354 0 000 .497.345.345 0 00.491 0L7.68 5.107a.345.345 0 01.492 0 .355.355 0 010 .498L6.697 7.097a.354.354 0 000 .497.345.345 0 00.492 0l.985-.997a.345.345 0 01.49.002.354.354 0 010 .498zm-.21-4.264a.354.354 0 010 .497l-.078.08a1.056 1.056 0 00-.492-.498l.079-.08a.345.345 0 01.491 0zm.492 1.989l-.079.08a1.05 1.05 0 00-.492-.498l.08-.08a.345.345 0 01.491 0 .355.355 0 010 .498zm-.937-3.519a.354.354 0 010 .496c-.043.045.257-.257-1.108 1.113a1.056 1.056 0 00-.49-.495l1.106-1.114a.345.345 0 01.492 0zm-5.463 3.03C3.584 3.283 6.018.831 6.042.806a.345.345 0 01.492 0c.13.131.135.337.02.474-3.25 3.282 1.53-1.553-4.29 4.333L2.09 4.365c.14.107.334.092.456-.032zM.98 2.066a.348.348 0 01.592-.228c.118.12.073.17.106.298l.228 1.63a1.037 1.037 0 00-.882-.458c-.007-.438-.021-.84-.044-1.242z"
              fill="currentColor"
            />
          </svg>
        </button>
        <span
          className="sc-bcXGCL hmaIsl cap-text"
        >
          10
        </span>
      </div>
      <div
        alwaysOpenInPortal={true}
        ariaLabel="global.report.submit"
        data-testid="Modal"
        hideCloseButton={true}
        onClose={[Function]}
        show={false}
        variantSize="xl"
      />
    </div>
  </div>,
]
`;

exports[`<DebateStepPageArgumentDrawer /> should render when open and viewer is the author 1`] = `
Array [
  <div
    className="sc-bcXGCL cgAMhb toasts-container toasts-container--top"
  />,
  <div
    data-testid="AnimatePresence"
  >
    <div
      className="sc-bcXGCL iLDPSx DetailDrawer__DetailDrawerInner-iBGoKl"
      style={
        Object {
          "opacity": 0,
          "transform": "translateX(100vw) translateZ(0)",
        }
      }
    >
      <div
        className="sc-bcXGCL ahQii"
      >
        <svg
          aria-hidden="true"
          className="sc-bcXGCL sc-eJDTud bPmEkv cpMilg cap-icon DetailDrawerHeader___StyledIcon-dNiWSR igkLcM detail__drawer--back-arow"
          fill="none"
          onClick={[MockFunction]}
          viewBox="0 0 12 6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 3H1M2.667 4.667L1 3l1.667-1.667"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1.5}
          />
        </svg>
        <div
          className="sc-bcXGCL gDgUrs"
        >
          <p
            className="sc-bcXGCL cUtTDY cap-text"
            style={
              Object {
                "marginLeft": 0,
                "marginTop": 0,
              }
            }
          >
            Agui le penseur
          </p>
          <div
            className="sc-bcXGCL kcdSuz sc-fXqqsB gKbWOe cap-tag"
            title=""
          >
            <h5
              className="sc-bcXGCL dUdqFq cap-heading"
            >
              global.for
            </h5>
          </div>
        </div>
      </div>
      <div
        className="sc-bcXGCL byWdSB"
        onClose={[MockFunction]}
      >
        <p
          className="sc-bcXGCL jMALnA cap-text"
        >
          coucou
        </p>
      </div>
      <div
        className="sc-bcXGCL cGdsLR"
        onClose={[MockFunction]}
      >
        <button
          aria-label="vote.add"
          className="sc-bcXGCL sc-laZSpB dDVxRe bGltjj cap-button"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-eJDTud kAyhOl cpMilg cap-icon"
            fill="none"
            viewBox="0 0 10 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M.001 4.365c.055.941.05 1.722.046 2.41-.01 1.826-.027 3.016 1.242 4.3a3.107 3.107 0 004.425 0l3.441-3.481a1.063 1.063 0 00-.288-1.698l.57-.577a1.065 1.065 0 000-1.492 1.037 1.037 0 00-.323-.222 1.063 1.063 0 00-.473-1.483A1.063 1.063 0 008.5.806c-.34-.344-.857-.4-1.255-.17A1.036 1.036 0 005.55.308c-.02.021-1.84 1.854-3.004 3.03L2.368 2.06a1.048 1.048 0 00-1.042-1.028A1.048 1.048 0 00.283 2.09c.028.478.043.96.048 1.502a1.05 1.05 0 00-.33.773zm8.663 2.732l-3.442 3.48c-.949.96-2.493.96-3.441 0C.726 9.511.73 8.598.742 6.78c.005-.696.01-1.483-.046-2.44a.348.348 0 01.593-.227c.118.12.073.17.105.298v.002l.28 1.996c.04.286.387.404.59.2.6-.608-.063.062 3.46-3.5a.345.345 0 01.481.01.354.354 0 010 .497L4.73 5.107a.354.354 0 000 .498.345.345 0 00.492 0l1.967-1.99a.345.345 0 01.491 0 .354.354 0 010 .498L5.714 6.102a.354.354 0 000 .497.345.345 0 00.491 0L7.68 5.107a.345.345 0 01.492 0 .355.355 0 010 .498L6.697 7.097a.354.354 0 000 .497.345.345 0 00.492 0l.985-.997a.345.345 0 01.49.002.354.354 0 010 .498zm-.21-4.264a.354.354 0 010 .497l-.078.08a1.056 1.056 0 00-.492-.498l.079-.08a.345.345 0 01.491 0zm.492 1.989l-.079.08a1.05 1.05 0 00-.492-.498l.08-.08a.345.345 0 01.491 0 .355.355 0 010 .498zm-.937-3.519a.354.354 0 010 .496c-.043.045.257-.257-1.108 1.113a1.056 1.056 0 00-.49-.495l1.106-1.114a.345.345 0 01.492 0zm-5.463 3.03C3.584 3.283 6.018.831 6.042.806a.345.345 0 01.492 0c.13.131.135.337.02.474-3.25 3.282 1.53-1.553-4.29 4.333L2.09 4.365c.14.107.334.092.456-.032zM.98 2.066a.348.348 0 01.592-.228c.118.12.073.17.106.298l.228 1.63a1.037 1.037 0 00-.882-.458c-.007-.438-.021-.84-.044-1.242z"
              fill="currentColor"
            />
          </svg>
        </button>
        <span
          className="sc-bcXGCL hmaIsl cap-text"
        >
          10
        </span>
      </div>
      <div
        alwaysOpenInPortal={true}
        ariaLabel="global.report.submit"
        data-testid="Modal"
        hideCloseButton={true}
        onClose={[Function]}
        show={false}
        variantSize="xl"
      />
    </div>
  </div>,
]
`;
